K2Bridge.Tests.UnitTests/Visitors/LuceneNet/LuceneMatchAllDocsVisitorTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace UnitTests.K2Bridge.Visitors.LuceneNet
+{
+    using System;
+    using global::K2Bridge.Models.Request.Queries;
+    using global::K2Bridge.Models.Request.Queries.LuceneNet;
+    using global::K2Bridge.Visitors;
+    using global::K2Bridge.Visitors.LuceneNet;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class LuceneMatchAllDocsVisitorTests
+    {
+        [TestCase]
+        public void Visit_WithNullQuery_ThrowsException()
+        {
+            var visitor = new LuceneVisitor();
+            Assert.That(
+                () => visitor.Visit((LuceneMatchAllDocsQuery)null),
+                Throws.TypeOf<ArgumentNullException>());
+        }
+
+        [TestCase]
+        public void Visit_WithInvalidQuery_ThrowsException()
+        {
+            var query = new LuceneMatchAllDocsQuery {
+                LuceneQuery = null,
+            };
+            var visitor = new LuceneVisitor();
+            Assert.That(
+                () => visitor.Visit(query),
+                Throws.TypeOf<IllegalClauseException>());
+        }
+
+        [TestCase(ExpectedResult = null)]
```

Comment:
why expected result is null?

Reply:
See K2Bridge/Visitors/QueryStringClauseVisitor.cs.

Match all matches everything, so we don't need to add anything to the kusto query.
I can replace it with String.Empty if you prefer